qt/Torrent.h

```diff
@@ -143,9 +144,9 @@ class TorrentHash
 
     QString toString() const
     {
-        char str[SHA_DIGEST_LENGTH * 2 + 1];
-        tr_sha1_to_hex(str, data_.data());
-        return QString::fromUtf8(str, SHA_DIGEST_LENGTH * 2);
+        char strbuf[TR_SHA1_DIGEST_LEN * 2 + 1];
+        tr_sha1_to_string(data_, strbuf);
+        return QString::fromUtf8(strbuf, SHA_DIGEST_LENGTH * 2);
```

Comment:
Use `std::size(data_)` / `std::size(tr_sha1_to_string)` instead of `TR_SHA1_DIGEST_LEN` / `SHA_DIGEST_LENGTH`, or use overload returning `std::string` (+`QString::fromStdString()`). Keeping `+1` isn't necessary now, `tr_sha1_to_string()` doesn't terminate the buffer.

Reply:
The overload resolves nicely to `return QString::fromStdString(tr_sha1_to_string(data_));`

That incurs the overhead of an extra heap allocation, but avoids this code having to make any assumptions about the width at all. On the face of it that's probably a good tradeoff, but I'll watch for this code next time I profile the Qt app.

Fixed c5414fdb0f5ad39f82a3557a15eceebe97e52f6b